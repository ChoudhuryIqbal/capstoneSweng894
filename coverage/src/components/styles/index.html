<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/styles</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> src/components/styles
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/0</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file empty" data-value="AccountAddress.styles.js"><a href="AccountAddress.styles.js.html">AccountAddress.styles.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file empty" data-value="AccountDetails.styles.js"><a href="AccountDetails.styles.js.html">AccountDetails.styles.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file empty" data-value="AccountForm.styles.js"><a href="AccountForm.styles.js.html">AccountForm.styles.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file empty" data-value="BottomBtn.styles.js"><a href="BottomBtn.styles.js.html">BottomBtn.styles.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file empty" data-value="ClearBtnSty.styles.js"><a href="ClearBtnSty.styles.js.html">ClearBtnSty.styles.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file empty" data-value="DeleteProfileBtn.js"><a href="DeleteProfileBtn.js.html">DeleteProfileBtn.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file empty" data-value="ErrorText.styles.js"><a href="ErrorText.styles.js.html">ErrorText.styles.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file empty" data-value="General.styles.js"><a href="General.styles.js.html">General.styles.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file empty" data-value="LogInBtn.styles.js"><a href="LogInBtn.styles.js.html">LogInBtn.styles.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file empty" data-value="PrefChoice.styles.js"><a href="PrefChoice.styles.js.html">PrefChoice.styles.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file empty" data-value="PrefResult.styles.js"><a href="PrefResult.styles.js.html">PrefResult.styles.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file empty" data-value="PrefTop.styles.js"><a href="PrefTop.styles.js.html">PrefTop.styles.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file empty" data-value="SignUpBtn.styles.js"><a href="SignUpBtn.styles.js.html">SignUpBtn.styles.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file empty" data-value="SkipNav.styles.js"><a href="SkipNav.styles.js.html">SkipNav.styles.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file empty" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Oct 26 2019 20:14:59 GMT-0400 (Eastern Daylight Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
<script src="..\..\..\block-navigation.js"></script>
</body>
</html>
