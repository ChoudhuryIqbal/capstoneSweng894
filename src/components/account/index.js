export * from './AccountButtons'
export * from './AccountDetails'
export * from './AccountAddress'
export * from './AccountForm'
export * from './AccountImage'
export * from './AccountImageUpload'
export * from './AccountLogIn'
export * from './AccountOptions'