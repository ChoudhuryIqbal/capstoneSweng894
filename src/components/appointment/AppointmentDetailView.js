import React, {useState} from 'react'
import { AppointmentView, AppointmentEditView } from '../appointment'
import { ButtonCustom } from '../common'
import LoginButton from '../styles/LoginButton.styles'

/**
 * Appointment Detail View - handle switch between edit and read-only view of appointment details
 * @param {*} props 
 */
const AppointmentDetailView = (props) => {
    const [editable, setEditable] = useState(false)
    const { address, state, city, zipCode, profile, preference, token,
        replaceItem, item, onDisplay} = props  
    
    const onEditClick = () =>{
        return (
            <ButtonCustom
                onPress={() => setEditable(!editable)}
                buttonStyle={LoginButton.buttonStyle}
                textStyle={LoginButton.textStyle}
            >
                {'Edit'}
            </ButtonCustom>
        )
    } 

    if(!editable){
        return (
            <AppointmentView
                onEditClick={onEditClick}
                item={item}
                address={address}
                city={city}
                state={state}
                zipCode={zipCode}
                profile={profile}
                preference={preference}
                token={token}
            />
        )
    } else {
        return (
            <AppointmentEditView
                onEditClick={onEditClick}
                item={item}
                address={address}
                city={city}
                state={state}
                zipCode={zipCode}
                profile={profile}
                preference={preference}
                token={token}
                replaceItem={replaceItem}
                onDisplay={onDisplay}
            />
        )
    }    
}

export { AppointmentDetailView }