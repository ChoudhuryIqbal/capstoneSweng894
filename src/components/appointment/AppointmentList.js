import React from 'react'
import { View, FlatList} from 'react-native'
import { SafeAreaView } from 'react-navigation'

const AppointmentList = (props) => {    
    return (
        <SafeAreaView>
            <View>
                <FlatList
                    contentContainerStyle={{ width: '98%'}}
                    scrollEnabled={props.scrollEnabled}
                    data={props.currentData}
                    extraData={props.extraData}
                    renderItem={props.renderItem}
                    ListHeaderComponent={props.listHeader}
                    ListEmptyComponent={props.listEmpty}
                    ItemSeparatorComponent={props.separator}
                    keyExtractor={props.keyExtractor}
                    onEndReachedThreshold={0.1}
                    initialScrollIndex={0}
                />
            </View>
        </SafeAreaView>
    )
}

export  {AppointmentList}