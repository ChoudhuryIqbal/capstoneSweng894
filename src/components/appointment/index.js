export * from './AppointmentList'
export * from './AppointmentItem'
export * from './AppointmentDetailView'
export * from './AppointmentEditView'
export * from './AppointmentView'
export * from './AppointmentRenderPickerField'
export * from './BusinessName'
export * from './Calendar'
export * from './Time'
export * from './Status'
export * from './Stylist'
export * from './Service'