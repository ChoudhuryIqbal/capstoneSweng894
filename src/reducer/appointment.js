import { actions } from '../actions/type'

const initalState = {
    upcomingAppointment: [],
    previousAppointment: [],
    errorMessage: '',
    loading: ''
}

export default (state = initalState, action ) => {
    switch (action.type) {
        case actions.SET_PREVIOUS_APPOINTMENT:{
            var a = {
                ...state,
                previousAppointment: action.payload.map(obj => ({...obj, 'listType': 'previous'})), paloading: false
            }

            // console.log('actions.SET_PREVIOUS_APPOINTMENT', a);
         
            return a
        }
        case actions.GET_PREVIOUS_APPOINTMENT_PENDING: {
            var b = {
                ...state,
                paloading: action.payload
            }

            // console.log('actions.GET_PREVIOUS_APPOINTMENT_PENDING', b)
            
            return b
        }
        case actions.GET_PREVIOUS_APPOINTMENT_FULFILLED: {
            var c = {
                ...state,
                previousAppointment: action.payload.map(obj => ({...obj, 'listType': 'previous'})), paloading: action.loading
            }

            // console.log('actions.GET_PREVIOUS_APPOINTMENT_FULFILLED', c);

            return c            
        }
        case actions.GET_PREVIOUS_APPOINTMENT_REJECTED:{
            
            // console.log('actions.GET_PREVIOUS_APPOINTMENT_REJECTED', action.loading);
            var d = {
                ...state,
                errorMessage: action.payload, paloading: action.loading
            }

            // console.log('actions.GET_PREVIOUS_APPOINTMENT_REJECTED', d);
            
            return d
        }
        case actions.SET_UPCOMING_APPOINTMENT:{
            var e = {
                ...state,
                upcomingAppointment: action.payload.map(obj => ({...obj, 'listType': 'upcoming'})), ualoading: false
            }

            // console.log('actions.SET_UPCOMING_APPOINTMENT', e);
         
            return e
        }
        case actions.GET_UPCOMING_APPOINTMENT_PENDING: {
            var f = {
                ...state,
                ualoading: action.payload
            }

            // console.log('actions.GET_UPCOMING_APPOINTMENT_PENDING', f)
            
            return f
        }
        case actions.GET_UPCOMING_APPOINTMENT_FULFILLED: {
            var g = {
                ...state,
                upcomingAppointment: action.payload.map(obj => ({...obj, 'listType': 'upcoming'})), ualoading: action.loading
            }

            // console.log('actions.GET_UPCOMING_APPOINTMENT_FULFILLED', g);

            return g            
        }
        case actions.GET_UPCOMING_APPOINTMENT_REJECTED:{
            var h = {
                ...state,
                errorMessage: action.payload, ualoading: action.loading
            }

            // console.log('actions.GET_UPCOMING_APPOINTMENT_REJECTED', h);
            
            return h
        }
        case actions.REPLACE_APPOINTMENT: {
            var list = action.listType == 'upcoming' ? state.upcomingAppointment : state.previousAppointment
            
            list.splice(list.indexOf(action.oldItem),0,action.newItem);

            var i = {
                ...state,
                upcomingAppointment: action.listType == 'upcoming' ? list : state.upcomingAppointment,
                previousAppointment: action.listType == 'upcoming' ? state.previousAppointment : list
            }

            // console.log('actions.REPLACE_APPOINTMENT', i);            
            
            return i
        }
        case actions.DELETE_APPOINTMENT: {
            var list = action.listType == 'upcoming' ? state.upcomingAppointment : state.previousAppointment
            
            list.splice(list.indexOf(action.deleteItem)-1,1);

            var k = {
                ...state,
                upcomingAppointment: action.listType == 'upcoming' ? list : state.upcomingAppointment,
                previousAppointment: action.listType == 'upcoming' ? state.previousAppointment : list
            }

            // console.log('actions.DELETE_APPOINTMENT', k);            
            
            return k
        }
        case actions.SET_ALL_APPOINTMENT:{
            var j = {
                ...state,
                allAppointment: action.payload.map(obj => ({...obj, 'listType': 'All'})), allloading: false
            }

            // console.log('actions.SET_UPCOMING_APPOINTMENT', j);
         
            return j
        }
        case actions.GET_ALL_APPOINTMENT_PENDING: {
            var k = {
                ...state,
                allloading: action.payload
            }

            // console.log('actions.GET_UPCOMING_APPOINTMENT_PENDING', k)
            
            return k
        }
        case actions.GET_ALL_APPOINTMENT_FULFILLED: {
            var l = {
                ...state,
                allAppointment: action.payload.map(obj => ({...obj, 'listType': 'all'})), allloading: action.loading
            }

            // console.log('actions.GET_UPCOMING_APPOINTMENT_FULFILLED', l);

            return l          
        }
        case actions.GET_ALL_APPOINTMENT_REJECTED:{
            var m = {
                ...state,
                errorMessage: action.payload, allloading: action.loading
            }

            // console.log('actions.GET_UPCOMING_APPOINTMENT_REJECTED', m);
            
            return m
        }
        default:
            return state;
    }
}

