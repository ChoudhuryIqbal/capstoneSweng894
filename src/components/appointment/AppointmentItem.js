import React from 'react'
import {View, Image, Text, TouchableOpacity } from 'react-native'
import date from 'date-and-time';
import 'date-and-time/plugin/ordinal'

import styles from '../styles/General.styles'

const AppointmentItem = (props) => {
    date.plugin('ordinal');
    
    return (
        <View>
            <TouchableOpacity onPress={props.onClick} onLongPress={props.onHoldClick}>
                <View style={styles.Row}>
                    <View style={styles.Row}>
                        <View style={{justifyContent: 'center', 
                                    paddingStart: 10, 
                                    paddingEnd: 1}}>
                            <Image
                                style={{
                                    width: 45,
                                    height: 45,
                                    borderRadius: 25
                                }}
                                source={require('../../image/Placeholder150.png')}
                            />  
                            {/* <Text style={{color: '#724FFD', fontSize: 35}}>{props.shopName.substring(0,1)}</Text> */}
                        </View>
                    </View>
                    <View style={styles.Column}>
                        <View style={{alignItems: 'flex-start'}}>
                            <Text style={{color: '#724FFD'}}>{props.shopName}</Text>
                            <Text style={{color: '#724FFD'}}>{props.service}</Text>
                        </View>
                    </View>
                    <View style={styles.Column}>
                        <View style={{alignSelf: 'flex-end'}}>
                            <Text style={{color: '#724FFD'}}>{date.format(date.parse(props.date, 'YYYY-MM-DD'), 'MMM DDD')+ '|' + props.time}</Text>
                            <Text style={{color: '#724FFD'}}>{props.status}</Text>
                        </View>
                    </View>
                </View>
            </TouchableOpacity>
        </View>
      
    )
}

export {AppointmentItem}