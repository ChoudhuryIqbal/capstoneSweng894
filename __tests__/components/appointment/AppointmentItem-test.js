import React from 'react'
import {AppointmentItem} from '../../../src/components/appointment'
import {create} from 'react-test-renderer';

describe('appointment item correctly render', () => {
    test('Matches the snapshot', () => {
        const onDetailClick = (item) => {}
        const onDetailHoldClickDelete = (itme) => {}

        const appointmentItem = create( 
            <AppointmentItem
                shopName={'Test'}
                service={'Test' }
                date={'2001-01-01'}
                time={'11:10:10'}
                status={'Test'}
                onClick={() => onDetailClick(item.item)}
                onHoldClick={() => onDetailHoldClickDelete(item.item)}
            />
        )
        expect(appointmentItem.toJSON()).toMatchSnapshot();
    })
})