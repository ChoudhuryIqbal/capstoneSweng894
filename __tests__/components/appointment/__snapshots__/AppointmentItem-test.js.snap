// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`appointment item correctly render Matches the snapshot 1`] = `
<View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "justifyContent": "center",
              "paddingEnd": 1,
              "paddingStart": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/image/hair-dresser.png",
              }
            }
            style={
              Object {
                "borderRadius": 25,
                "height": 45,
                "width": 45,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flex": 1,
            "flexBasis": 90,
            "flexDirection": "column",
            "justifyContent": "center",
            "margin": 10,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#724FFD",
              }
            }
          >
            Test
          </Text>
          <Text
            style={
              Object {
                "color": "#724FFD",
              }
            }
          >
            Test
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flex": 1,
            "flexBasis": 90,
            "flexDirection": "column",
            "justifyContent": "center",
            "margin": 10,
          }
        }
      >
        <View
          style={
            Object {
              "alignSelf": "flex-end",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#724FFD",
              }
            }
          >
            Jan 1st|11:10:10
          </Text>
          <Text
            style={
              Object {
                "color": "#724FFD",
              }
            }
          >
            Test
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
